### ISSUE TITLE
[BUG] 15 minute cooldown no points being earned after 4 searches

### ISSUE THREAD

**Issue by icescoop12** [charlesbel/Microsoft-Rewards-Farmer#415]
### Before submitting a bug report...

- [X] This bug wasn't already reported.
(I have checked every bug report on GitHub)

### Title

- [X] The title is no longer "[BUG] Title" and I edited it with the right error name.

### Describe the bug

due to MS implementing th 15 minute cooldown between every 3 searches
see [here](https://www.reddit.com/r/MicrosoftRewards/comments/18hakg9/15_minutes_cool_down_permanent/?utm_source=share&utm_medium=web2x&context=3)
can we adjust the bot to do 3 searches on either desktop or mobile at a time then run it on task schedule so pc sleeps between
what needs adjusting in the script to do just 3 searches only no matter point tally until its no more to gain?

### Copy and paste your error

none

### Screenshots

none

### Value of dashboard variable

none

**Comment by ExtremeMemes** [comment]
Try adjusting the sleep time in the searches.py file on line 85.

I put mine from ~6 minutes to ~15 minutes, it will chose a random moment to run it. Might take a while, but will get the job done. 

`time.sleep(random.randint(360, 900))`

**Comment by comcord14** [comment]
I added the following to the for loop that starts on line 58 in searches.py:

`if i % 4 == 0:
     time.sleep(900)`

This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).

**Comment by ExtremeMemes** [comment]
> I added the following to the for loop that starts on line 58 in searches.py:
> 
> `if i % 4 == 0: time.sleep(900)`
> 
> This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).

yeah this works! you should make a pr

**Comment by caaaab** [comment]
This is what I made and I think it works. Fills the console every minute which I'm fine with but might get obnoxious for others. I have klept0's fork but the base should look similar if not the same

src/searches.py in the bingSearches function, new code comes right after the for loop

```
i = 0
        for word in search_terms:
            if i != 0 and i % 3 == 0:
                logging.info(
                    f"[BING] [PAUSE] Searched {i} times (multiple of 3), pausing for 15 minutes."
                )
                time.sleep(60)
                for i in range(14):
                    logging.info(
                        f"[BING] [PAUSE] {14-i} minutes remaining."
                    )
                    time.sleep(60)
                logging.info(
                    "[BING] [UNPAUSE] Resuming searches..."
                )
            i += 1
```

**Comment by icescoop12** [comment]
> I added the following to the for loop that starts on line 58 in searches.py:
> 
> `if i % 4 == 0: time.sleep(900)`
> 
> This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).

okay tried that thanks

but is there a way to make the program just do 4 searches only,this way we can run task schedule or a cron job and not having it running countless hours?
also to add an extra 2 searches as earning starts at your third search

**Comment by Forsefrits** [comment]
> I added the following to the for loop that starts on line 58 in searches.py:
> 
> `if i % 4 == 0: time.sleep(900)`
> 
> This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).

Sorry for being stupid but do I edit or add this overthere? It looks like a brilliant fix instead of having to run the program for a whole day with the 15m pauses!
Line 58 has "for word in search_terms:".
Just put an extra line with this under/above it?

**Comment by agustinscaz** [comment]
> > I added the following to the for loop that starts on line 58 in searches.py:
> > `if i % 4 == 0: time.sleep(900)`
> > This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).
> 
> Sorry for being stupid but do I edit or add this overthere? It looks like a brilliant fix instead of having to run the program for a whole day with the 15m pauses! Line 58 has "for word in search_terms:". Just put an extra line with this under/above it?

It should be like this:
```
        for word in search_terms:
            if i % 4 == 0: time.sleep(900)
            i += 1
```

**Comment by Forsefrits** [comment]
> > > I added the following to the for loop that starts on line 58 in searches.py:
> > > `if i % 4 == 0: time.sleep(900)`
> > > This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).
> > 
> > 
> > Sorry for being stupid but do I edit or add this overthere? It looks like a brilliant fix instead of having to run the program for a whole day with the 15m pauses! Line 58 has "for word in search_terms:". Just put an extra line with this under/above it?
> 
> It should be like this:
> 
> ```
>         for word in search_terms:
>             if i % 4 == 0: time.sleep(900)
>             i += 1
> ```

It's working thanks! 
It's doing 78 searches now btw. Where can I change that to 30 as each search gives 3 points?

**Comment by EO-sysadmin** [comment]
I added the line above and I'm getting stuck at the "Starting Desktop Edge Bing Searches". I run using the -v so I can see what's happening and it's just sitting at the rewards page.

**Comment by Forsefrits** [comment]
> I added the line above and I'm getting stuck at the "Starting Desktop Edge Bing Searches". I run using the -v so I can see what's happening and it's just sitting at the rewards page.

Had this as well, it was because of the higher delay that I did set `time.sleep(random.randint(360, 900))` this causes the program to wait 3-15 minutes
The delay can be between 10 and 15 as default, but for Microsoft's detection I have set this to 10 and 40 just to be safe (hopefully).

**Comment by Forsefrits** [comment]
Anyone who knows where I can make the program to do less searches?
90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.

**Comment by icescoop12** [comment]
> Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.

take a look here fixed it for me, soluion at the end
#401

**Comment by Forsefrits** [comment]
> > Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.
> 
> take a look here fixed it for me, soluion at the end #401

Thanks! Have to test it tomorrow to be sure but it cancels searching when started now, because of completion. It did not do that before, so it looks promising!

**Comment by Forsefrits** [comment]
> > > Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.
> > 
> > 
> > take a look here fixed it for me, soluion at the end #401
> 
> Thanks! Have to test it tomorrow to be sure but it cancels searching when started now, because of completion. It did not do that before, so it looks promising!

It works like a charm!

Next thing is that every time I use this tool, Microsoft is temporarily restricting my Bing Rewards account, I cannot redeem anymore.. 
Is there anything that gives away that I'm using this tool, which therefore causes an automatic ban of my Rewards account?
It recovers after a while of doing "human" searches. But annoying when I want to claim a giftcard and it does not work.

I know that the tool maker is not responsible for it and I am not looking for someone to blame, but I want to understand how Microsoft detects this and perhaps try to solve that detection.

**Comment by icescoop12** [comment]
> > > > Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.
> > > 
> > > 
> > > take a look here fixed it for me, soluion at the end #401
> > 
> > 
> > Thanks! Have to test it tomorrow to be sure but it cancels searching when started now, because of completion. It did not do that before, so it looks promising!
> 
> It works like a charm!
> 
> Next thing is that every time I use this tool, Microsoft is temporarily restricting my Bing Rewards account, I cannot redeem anymore.. Is there anything that gives away that I'm using this tool, which therefore causes an automatic ban of my Rewards account? It recovers after a while of doing "human" searches. But annoying when I want to claim a giftcard and it does not work.
> 
> I know that the tool maker is not responsible for it and I am not looking for someone to blame, but I want to understand how Microsoft detects this and perhaps try to solve that detection.

yep same has happened to me ,going to have to live with this ,i think its undetected chromedriver getting detected

**Comment by harukodi** [comment]
Anyone tried pyautogui instead?

**Comment by Forsefrits** [comment]
> > > > > Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.
> > > > 
> > > > 
> > > > take a look here fixed it for me, soluion at the end #401
> > > 
> > > 
> > > Thanks! Have to test it tomorrow to be sure but it cancels searching when started now, because of completion. It did not do that before, so it looks promising!
> > 
> > 
> > It works like a charm!
> > Next thing is that every time I use this tool, Microsoft is temporarily restricting my Bing Rewards account, I cannot redeem anymore.. Is there anything that gives away that I'm using this tool, which therefore causes an automatic ban of my Rewards account? It recovers after a while of doing "human" searches. But annoying when I want to claim a giftcard and it does not work.
> > I know that the tool maker is not responsible for it and I am not looking for someone to blame, but I want to understand how Microsoft detects this and perhaps try to solve that detection.
> 
> yep same has happened to me ,going to have to live with this ,i think its undetected chromedriver getting detected

That sounds like a logical reason for detection indeed. Afraid we have to live with it, yup.



> Anyone tried pyautogui instead?

Nope, did you and did it help?

**Comment by harukodi** [comment]
> > > > > > Anyone who knows where I can make the program to do less searches? 90 and 60 is a lot and takes nearly a whole day with 15 minutes delay per 4 searches with the changes suggested.
> > > > > 
> > > > > 
> > > > > take a look here fixed it for me, soluion at the end #401
> > > > 
> > > > 
> > > > Thanks! Have to test it tomorrow to be sure but it cancels searching when started now, because of completion. It did not do that before, so it looks promising!
> > > 
> > > 
> > > It works like a charm!
> > > Next thing is that every time I use this tool, Microsoft is temporarily restricting my Bing Rewards account, I cannot redeem anymore.. Is there anything that gives away that I'm using this tool, which therefore causes an automatic ban of my Rewards account? It recovers after a while of doing "human" searches. But annoying when I want to claim a giftcard and it does not work.
> > > I know that the tool maker is not responsible for it and I am not looking for someone to blame, but I want to understand how Microsoft detects this and perhaps try to solve that detection.
> > 
> > 
> > yep same has happened to me ,going to have to live with this ,i think its undetected chromedriver getting detected
> 
> That sounds like a logical reason for detection indeed. Afraid we have to live with it, yup.
> 
> > Anyone tried pyautogui instead?
> 
> Nope, did you and did it help?

Yes, I've successfully implemented it, and it's functioning as expected. Additionally, I incorporated time.sleep(random.randint(1, 200)). To gain further insights, I created a diagram illustrating the distribution of random values, examining the pattern and assessing its recurrence. :)

**Comment by Forsefrits** [comment]
> 
> Yes, I've successfully implemented it, and it's functioning as expected. Additionally, I incorporated time.sleep(random.randint(1, 200)). To gain further insights, I created a diagram illustrating the distribution of random values, examining the pattern and assessing its recurrence. :)

Is it hard to implement that? :)

**Comment by harukodi** [comment]
It could be done in under 30 min but you will have to try playing around with the timer and such. Don't they use AI now to detect bots? Probably that's why people use random to make the pattern more random. Am not trying to automate and abuse the system with multiple accounts i just want the help with the mobile searches. Sadly my solution uses GUI and can not run in headless mode.

**Comment by harukodi** [comment]
If you are trying to make your own version you could also use an API to fetch random words or so!

**Comment by Forsefrits** [comment]
> It could be done in under 30 min but you will have to try playing around with the timer and such. Don't they use AI now to detect bots? Probably that's why people use random to make the pattern more random. Am not trying to automate and abuse the system with multiple accounts i just want the help with the mobile searches. Sadly my solution uses GUI and can not run in headless mode.

Yeah that's the weird part, I have put randoms in it and yet they detect this bot right away :(

**Comment by harukodi** [comment]
> > It could be done in under 30 min but you will have to try playing around with the timer and such. Don't they use AI now to detect bots? Probably that's why people use random to make the pattern more random. Am not trying to automate and abuse the system with multiple accounts i just want the help with the mobile searches. Sadly my solution uses GUI and can not run in headless mode.
> 
> Yeah that's the weird part, I have put randoms in it and yet they detect this bot right away :( 

Do like this set the time.sleep(400 + random.randint(1, 110) now am having problems with phone searches but that problem seems to be when i even try to use a phone and not the edge emulation i will see how to fix that in this couple of days

**Comment by crazysapien** [comment]
> I added the following to the for loop that starts on line 58 in searches.py:
> 
> `if i % 4 == 0: time.sleep(900)`
> 
> This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).

what will happen if i use multiple accounts. does the program stops for 15 minutes or does the searches start for other account ?

**Comment by harukodi** [comment]
> > I added the following to the for loop that starts on line 58 in searches.py:
> > 
> > `if i % 4 == 0: time.sleep(900)`
> > 
> > This will make the program sleep for 15 minutes after every 4 searches (I only get the cooldown after 4 searches).
> 
> what will happen if i use multiple accounts. does the program stops for 15 minutes or does the searches start for other account ?

Hi it depends if this software runs in threads of the selenium lib. You can have multiple accounts but if it runs Vertically Horizontally am not really sure. But you should be able to run it Horizontally. Microsoft may notice that multiple accounts are searching at the same time and flag you. I got one account banned and i actually didn't run it Horizontally. I ran my own script Vertically. Just test your way. I would recommend just using it for one account that you may also use normally! 😊
